pallets/corporate-actions/src/distribution/mod.rs

```diff
@@ -348,13 +347,15 @@ decl_module! {
         /// - `NotExpired` if `now < expiry`.
         #[weight = <T as Config>::DistWeightInfo::reclaim()]
         pub fn reclaim(origin, ca_id: CAId) {
-            // Ensure DID is the dist creator, they haven't reclaimed, and that expiry has passed.
-            let did = <Identity<T>>::ensure_perms(origin)?;
             let dist = Self::ensure_distribution_exists(ca_id)?;
-            ensure!(did == dist.from.did, Error::<T>::NotDistributionCreator);
-            let did = did.for_event();
             ensure!(!dist.reclaimed, Error::<T>::AlreadyReclaimed);
             ensure!(expired(dist.expires_at, <Checkpoint<T>>::now_unix()), Error::<T>::NotExpired);
+            let PermissionedCallOriginData {
+                primary_did: caa,
+                secondary_key,
+                ..
+            } = <ExternalAgents<T>>::ensure_agent_asset_perms(origin.clone(), ca_id.ticker)?;
```

Comment:
Personally I like to see this check at the start of the extrinsic.  Most extrinsics do this check first.

Reply:
understood, would make the change.